pkg/es/config/config.go

```diff
@@ -70,34 +81,54 @@ func (p IndexPrefix) Apply(indexName string) string {
-	ServiceCacheTTL          time.Duration          `mapstructure:"service_cache_ttl"`
-	AdaptiveSamplingLookback time.Duration          `mapstructure:"-"`
-	Tags                     TagsAsFields           `mapstructure:"tags_as_fields"`
-	Enabled                  bool                   `mapstructure:"-"`
-	TLS                      configtls.ClientConfig `mapstructure:"tls"`
-	UseReadWriteAliases      bool                   `mapstructure:"use_aliases"`
-	CreateIndexTemplates     bool                   `mapstructure:"create_mappings"`
-	UseILM                   bool                   `mapstructure:"use_ilm"`
-	Version                  uint                   `mapstructure:"version"`
-	LogLevel                 string                 `mapstructure:"log_level"`
-	SendGetBodyAs            string                 `mapstructure:"send_get_body_as"`
+	// Servers is a list of Elasticsearch servers. The strings must must contain full URLs
+	// (i.e. http://localhost:9200).
+	Servers []string `mapstructure:"server_urls" valid:"required,url"`
+	// RemoteReadClusters is a list of Elasticsearch remote cluster names for cross-cluster
+	// querying.
+	RemoteReadClusters []string       `mapstructure:"remote_read_clusters"`
+	Authentication     Authentication `mapstructure:"auth"`
+	// TLS contains the TLS configuration for the connection to the ElasticSearch clusters.
+	TLS      configtls.ClientConfig `mapstructure:"tls"`
+	Sniffing Sniffing               `mapstructure:"sniffing"`
+	// SendGetBodyAs is the HTTP verb to use for requests that contain a body.
+	SendGetBodyAs string `mapstructure:"send_get_body_as"`
+	// Timeout contains the timeout used for queries. A timeout of zero means no timeout.
+	Timeout time.Duration `mapstructure:"timeout"`
+
+	BulkProcessing BulkProcessing `mapstructure:"bulk_processing"`
+	// Version contains the major Elasticsearch version. If this field is not specified,
+	// the value will be auto-detected from Elasticsearch.
+	Version uint `mapstructure:"version"`
+	// LogLevel contains the Elasticsearch client log-level. Valid values for this field
+	// are: [debug, info, error]
+	LogLevel string `mapstructure:"log_level"`
+
+	Indices Indices `mapstructure:"indices"`
+	// UseReadWriteAliases, if set to true, will use read and write aliases for indices.
+	// Use this option with Elasticsearch rollover API. It requires an external component
+	// to create aliases before startup and then performing its management.
+	UseReadWriteAliases bool `mapstructure:"use_aliases"`
+	// CreateIndexTemplates, if set to true, creates index templates at application startup.
+	// This configuration should be set to false when templates are installed manually.
+	CreateIndexTemplates bool `mapstructure:"create_mappings"`
+	// Option to enable Index Lifecycle Management (ILM) for Jaeger span and service indices.
+	// Read more about ILM at
+	// https://www.elastic.co/guide/en/elasticsearch/reference/current/index-lifecycle-management.html.
```

Comment:
nit: we should be pointing to our docs https://www.jaegertracing.io/docs/deployment/#enabling-ilm-support